init/functions

```diff
@@ -202,8 +212,14 @@ initialize_home() {
         homedirwithouthome=${homedir#/home/}
         if ! test -d "$home_root/$homedirwithouthome" || [ "$mode" = "unconditionally" ] ; then
             echo "initialize_home: populating $mode $home_root/$homedirwithouthome from /etc/skel" >&2
-            mkdir -p "$home_root/$homedirwithouthome"
-            if test -d /sys/fs/selinux; then enable_selinux="Z"; else enable_selinux=''; fi
+            if [ "$mode" = unconditionally ]; then
+                mkdir "-p${enable_selinux}" -- "$home_root/$homedirwithouthome" || return 73
+            else
+                case $homedirwithouthome in
+                (*/*) mkdir "-p${enable_selinux}" -- "$home_root/${homedirwithouthome%/*}";;
+                esac
+                mkdir "${enable_selinux+-Z}" -- "$home_root/$homedirwithouthome" || return 73
```

Comment:
Why splitting this into two mkdir calls (for which you needed to split this into two branches and extra pattern matching)? If you want this to fail if directory already exists, there is no point in that - this part is under `if ! test -d "$home_root/$homedirwithouthome"` (and this script doesn't really need to worry about racing against anything else accessing `/rw`).
But if there is some other reason, please explain it in the commit message.